core/src/main/scala/io/github/finagle/Serial.scala

```diff
@@ -85,9 +85,22 @@ trait Serial {
     Server[Req, Rep](reqCodec, repCodec)
 
   /**
-   * Returns an instance of [[com.twitter.finagle.Client]] with
-   * [[com.twitter.finagle.Server]] of concrete serial protocol from ''Req'' to
-   * ''Rep''.
+   * A concrete Serial protocol instance from 'Req'' to ''Rep''.
+   */
+  trait Protocol[Req, Rep] extends finagle.Client[Req, Rep] with finagle.Server[Req, Rep] {
+    /**
+     * A convenience method that creates a server from a function.
+     */
+    def serveFunction(addr: SocketAddress)(f: Req => Rep): ListeningServer = serve(
```

Comment:
Why is this not in `finagle-core`? Seems pretty useful for me. Do you think we don't need `serveAsyncFunction(f: Req => Future[Rep])`?